src/foam/nanos/auth/ServiceProviderAwareSupport.js

```diff
@@ -101,7 +101,7 @@ returning true if the spid or context users spid matches the current object.`,
       if ( obj != null &&
            obj instanceof ServiceProviderAware ) {
         ServiceProviderAware sp = (ServiceProviderAware) obj;
-        return sp.getSpid().startsWith(spid) ||
+        return sp.getSpid().equals("*") || sp.getSpid().startsWith(spid) ||
```

Comment:
This will be the least common case, so perhaps swap the order of the first two tests.

Reply:
Fixed in https://github.com/foam-framework/foam2/pull/4148/commits/6026f7ada2b1a6f08141995c702099d6198db7d6.